dev/src/index.ts

```diff
@@ -1416,11 +1423,15 @@ export class Firestore {
     const attempts = allowRetries ? MAX_REQUEST_RETRIES : 1;
     const callOptions = this.createCallOptions();
 
-    const gapicClient = this._clientPool.acquire();
-    const releaser = this._clientPool.createReleaser(gapicClient);
+    const result = new Deferred<NodeJS.ReadWriteStream>();
+
+    this._clientPool.run(gapicClient => {
+      // While we return the stream to the callee early, we don't want to
+      // release the GAPIC client until the callee has finished processing the
+      // stream.
+      const lifetime = new Deferred<void>();
 
-    return this._retry(attempts, requestTag, () => {
-      return Promise.resolve().then(() => {
+      this._retry(attempts, requestTag, async () => {
```

Comment:
Isn't this subtly different from the original code? Doesn't `Promise.resolve().then(...)` give the runtime the opportunity to go to the next tick? Did the original code intend that?